### PR TITLE
Javascript javascript3 week3

### PR DESCRIPTION
**Pull request by RemyaManoharan**

### CODE REVIEW

**Comment by shpomp** [comment]
@RemyaManoharan  this has a lot of files committed, could you specify those that should be checked by the mentors?

**Comment by RemyaManoharan** [comment]
> @RemyaManoharan this has a lot of files committed, could you specify those that should be checked by the mentors?

No..only last two files..js3-week3 hw  and js file for API fetch has to reviewd..Other files are from js2-week1 which are already reviewed. Its by mistake I commit all files of week1.

File: javascript/javascript3/week3/app.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import env from "./env.json" assert { type: "json" };
```
Comment (chezzoba) [inline javascript/javascript3/week3/app.js:1]: Nicely done

File: javascript/javascript3/week3/app.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+// const saveBtnTag = document.getElementById("saveBtn");
+const urltag = document.getElementById("url");
+const message = document.getElementById("message");
+const screenshotList = document.getElementById("screenshotList");
+const screenshotTag = document.getElementsByClassName("screenshot-wrapper");
+let screenshotUrl;
+
+// function to get screenshot
+const getData = async (urltag) => {
+  const url = `https://website-screenshot6.p.rapidapi.com/screenshot?url=${urltag}&width=1000&height=800&fullscreen=false`;
```
Comment (chezzoba) [inline javascript/javascript3/week3/app.js:13]: You can get the screensize dynamically by using the screen object so you don't have to hardcode these:
```javascript
const screenWidth = window.screen.width;
const screenHeight = window.screen.height;

console.log("Screen width: " + screenWidth);
console.log("Screen height: " + screenHeight);
```

File: javascript/javascript3/week3/app.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  const options = {
+    method: "GET",
+    headers: {
+      "X-RapidAPI-Key": env.apikey,
```
Comment (chezzoba) [inline javascript/javascript3/week3/app.js:17]: Very nice

File: javascript/javascript3/week3/app.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+});
+
+// function to display screenshot from api
+function displayScreenShot(screenshotUrl) {
+  console.log(screenshotUrl);
+  const image = document.querySelector(".screenshot");
+  image.src = screenshotUrl;
+  image.alt = "website-screenshot";
+  const screenshotWrapper = document.querySelector(".screenshot-wrapper");
+  const saveButton = document.createElement("button");
```
Comment (chezzoba) [inline javascript/javascript3/week3/app.js:47]: Very well done

File: javascript/javascript3/week3/app.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+function showSavingMessage() {
+  message.innerText = "Your screen shot has saved to server";
+}
+function showPubMessage() {
+  message.innerText = "Please wait... Publishing your screenshot";
+}
+
+// save button click event
+document.addEventListener("click", async (e) => {
+  if (e.target && e.target.id === "saveBtn") {
```
Comment (chezzoba) [inline javascript/javascript3/week3/app.js:62]: Check [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) out. Now you can do this:

```javascript
if (e.target?.id === "saveBtn")
```

File: javascript/javascript3/week3/app.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    console.log(error);
+  }
+};
+
+function showSavedUrlList(savedData) {
+  console.log(savedData);
+  const screenshots = document.querySelector(".screenshots");
+  screenshots.style.display = "block";
+  const screenshotItem = document.createElement("li");
+  screenshotItem.innerHTML = `
```
Comment (chezzoba) [inline javascript/javascript3/week3/app.js:94]: You got lazy there:

```javascript
// Create the elements
const link = document.createElement('a');
link.href = savedData.screenshotUrl;
link.target = '_blank';
link.textContent = urltag.value;

const button = document.createElement('button');
button.classList.add('deleteButton');
button.dataset.id = savedData._id;
button.textContent = 'Delete';

// Append the elements to the screenshotItem
screenshotItem.appendChild(link);
screenshotItem.appendChild(button);
```

**Review by chezzoba — APPROVED** [review]
Really great job! You deserve a sticker for how well done this is.

File: javascript/javascript3/week3/app.js (hunk 6 above)
Comment (RemyaManoharan) [reply javascript/javascript3/week3/app.js:94]: Thanks for the feedback

**Pull request merged**